lib/bitcrowd_ecto/schema.ex

```diff
@@ -46,7 +46,36 @@ defmodule BitcrowdEcto.Schema do
       @timestamps_opts [type: :utc_datetime_usec]
 
       @type t :: %__MODULE__{}
-      @type id :: binary()
+      @type id :: binary
     end
   end
+
+  @doc """
+  Safely converts a string into an enum member atom. Returns nil if conversion is not posssible.
+  """
+  @doc since: "0.9.0"
+  @spec safe_cast_enum_member(schema :: module, field :: atom, value :: any) :: term | nil
```

Comment:
Is `any` not to generic for `value`? According to the code it's either an atom or a string :atom:

Reply:
:) well technically you can pass any type and the code won't break.. but I guess you're right

Reply:
Probably we could at least allow `nil` as well, for optional values that pass this function.
```suggestion
  @spec safe_cast_enum_member(schema :: module, field :: atom, value :: atom() | String.t() | nil) :: term | nil
```

Reply:
:D as said, technically we allow anything here. Maybe I'll just revert this to `any`.

Reply:
Also helps keeping the spec in one line :)